### PR TITLE
Set active timestamp to 0 if there is no valid active operational dataset

### LINKED ISSUE #779: Certification: Missing Active TimeStamp TLV in data response packet
Certification test 8.3.1(Leader) fails since there is no Active TimeStamp TLV in data response packet.

[Leader_8_3_1.zip](https://github.com/openthread/openthread/files/518265/Leader_8_3_1.zip)

### PR DESCRIPTION
**Pull request by chshu**
Resolves #779.

With the help of #743 , we have passed certification test 8.3.1(Leader):
[Leader_8_3_1.zip](https://github.com/openthread/openthread/files/519253/Leader_8_3_1.zip)

### CODE REVIEW

File: src/core/thread/mle.cpp [hunk 1]
```diff
@@ -2101,10 +2108,11 @@ ThreadError Mle::HandleChildIdResponse(const Message &aMessage, const Ip6::Messa
     SuccessOrExit(error = Tlv::GetTlv(aMessage, Tlv::kNetworkData, sizeof(networkData), networkData));
 
     // Active Timestamp
-    if (Tlv::GetTlv(aMessage, Tlv::kActiveTimestamp, sizeof(activeTimestamp), activeTimestamp) == kThreadError_None)
-    {
-        VerifyOrExit(activeTimestamp.IsValid(), error = kThreadError_Parse);
+    SuccessOrExit(error = Tlv::GetTlv(aMessage, Tlv::kActiveTimestamp, sizeof(activeTimestamp), activeTimestamp));
```
Comment (xiaom-GitHub) [inline src/core/thread/mle.cpp]: Not sure whether or not MLE Active Timestamp TLV is necessary in ChildIdResponse message. If it's optional, using `SuccessOrExit()` will miss the following check.(i.e. send Address Solicit to request Router ID).

Comment (jwhui) [inline src/core/thread/mle.cpp]: The spec seems to indicate that the Active Timestamp TLV is not required in the Child ID Response message.  In particular, if the Child ID Request indicates that the joining child already has the proper Active Timestamp, then the Child ID Response does not need to include the Active Timestamp TLV.

Comment (chshu) [inline src/core/thread/mle.cpp]: Yes, the current implementation is a little not specification compliant. I will update it, Thanks.

**Comment by chshu** [comment]
@jwhui @xiaom-GitHub , I have updated this PR, please help to review it again, thanks.

**Comment by chshu** [comment]
This PR only resolves #779 and helps to pass certification test 8.3.1(Leader). 

I have pushed another separate PR #794 to fix the Active Timestamp TLV in Child ID Response issue, which still has some travis check failures need to address.

**Comment by jwhui** [comment]
Okay, I'm fine with merging this now since it gets us passing tests.

**Pull request merged**